code/game/jobs/job_objective.dm

```diff
@@ -1,69 +1,74 @@
-/datum/mind/var/list/job_objectives = list()
-
-#define FINDJOBTASK_DEFAULT_NEW 1 // Make a new task of this type if one can't be found.
-/datum/mind/proc/findJobTask(typepath, options = 0)
-	var/datum/job_objective/task = locate(typepath) in job_objectives
-	if(!istype(task,typepath))
-		if(options & FINDJOBTASK_DEFAULT_NEW)
-			task = new typepath()
-			job_objectives += task
-	return task
+/datum/mind/proc/find_job_task(typepath)
+	return locate(typepath) in job_objectives
 
 /datum/job_objective
-	var/datum/mind/owner = null			//Who owns the objective.
-	var/completed = 0					//currently only used for custom objectives.
-	var/per_unit = 0
-	var/units_completed = 0
-	var/units_compensated = 0 // Shit paid for
-	var/units_requested = INFINITY
-	var/completion_payment = 0			// Credits paid to owner when completed
+	///Fluff name of the objective
+	var/objective_name = "Placeholder Objective"
+	///short description of what the objective entails, should include how to complete the objective if not immediately clear
+	var/description = "A coder fucked up, one can finish this objective by making an issue report on the GitHub (Top-right of the screen has a button for it)"
+	///Who owns the objective
+	var/datum/mind/owner
+	///Which money account this objective is tied to, makes life easier during payout (signficantly)
+	var/datum/money_account/owner_account
+
+	///Has this objective been completed?
+	var/completed = FALSE
+	///Has the owner been rewarded yet?
+	var/payout_given = FALSE
+
+	///does this objective give out monetary rewards upon completion?
+	var/gives_payout = FALSE
+	///how many credits are awarded upon completion of job objective
+	var/completion_payment = 0
 
 /datum/job_objective/New(datum/mind/new_owner)
 	owner = new_owner
 	owner.job_objectives += src
 
+/datum/job_objective/proc/set_owner_account(datum/money_account/account) //needed for GC'ing
+	owner_account = account
+	RegisterSignal(owner_account, COMSIG_PARENT_QDELETING, PROC_REF(clear_owner_account))
 
-/datum/job_objective/proc/get_description()
-	var/desc = "Placeholder Objective"
-	return desc
-
-/datum/job_objective/proc/unit_completed(count=1)
-	units_completed += count
+/datum/job_objective/proc/clear_owner_account() //needed for GC'ing
+	UnregisterSignal(owner_account, COMSIG_PARENT_QDELETING)
+	owner_account = null
 
 /datum/job_objective/proc/is_completed()
 	if(!completed)
 		completed = check_for_completion()
 	return completed
 
 /datum/job_objective/proc/check_for_completion()
-	if(per_unit)
-		if(units_completed > 0)
-			return 1
-	return 0
+	//please override me :)
+	CRASH("check_for_completion() not overridden on [type]")
```

Comment:
IMO we don't need CI checks for stuff like this. I would say stick with the `CRASH`.
1. Because we need to add a new unique check every time this situation happens 
2. Adding things to CI is more complex and people may not know how (I don't even know how to do this)